clients/go/entities/job_test.go

```diff
@@ -15,72 +15,81 @@
 package entities
 
 import (
-	"github.com/zeebe-io/zeebe/clients/go/pb"
-	"reflect"
 	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/zeebe-io/zeebe/clients/go/pb"
 )
 
-type MyType struct {
+type testType struct {
 	Foo   string
 	Hello string
 }
 
 var (
-	expectedJson      = "{\"foo\": \"bar\", \"hello\": \"world\"}"
-	expectedJsonAsMap = map[string]string{
-		"foo":   "bar",
-		"hello": "world",
-	}
-	expectedJsonAsMapVariables = map[string]interface{}{
-		"foo":   "bar",
-		"hello": "world",
-	}
-	expectedJsonAsStruct = MyType{
+	job = Job{pb.ActivatedJob{
+		CustomHeaders: `{"foo": "bar", "hello": "world"}`,
+		Variables:     `{"foo": "bar", "hello": "world"}`,
+	}}
+	wantStruct = testType{
 		Foo:   "bar",
 		Hello: "world",
 	}
-	job = Job{pb.ActivatedJob{
-		CustomHeaders: expectedJson,
-		Variables:     expectedJson,
-	}}
 )
 
 func TestJob_GetVariablesAsMap(t *testing.T) {
-	variables, err := job.GetVariablesAsMap()
+	got, err := job.GetVariablesAsMap()
 	if err != nil {
-		t.Error("Failed to get variables as map", err)
+		t.Errorf("job.GetVariablesAsMap() = %v", err)
```

Comment:
I understand preferring `Error` to `Fail` but in this case `got` is invalid so we can fail here.

Reply:
Good point.